packages/core/src/animation/Animator.ts

```diff
@@ -418,7 +421,22 @@ export class Animator extends Component {
           return scale;
       }
     }
-    return value;
+
+    switch (curve._valueType) {
+      case InterpolableValueType.Float:
+        return value;
+      case InterpolableValueType.FloatArray:
+        for (let i = 0, n = (<Float32Array>value).length; i < n; i++) {
+          out[i] = value[i];
+        }
+        return out;
+      case InterpolableValueType.Vector2:
+      case InterpolableValueType.Vector3:
+      case InterpolableValueType.Vector4:
+      case InterpolableValueType.Quaternion:
+        (value as IClone).cloneTo(out as IClone);
```

Comment:
We can set out  param to `curve.evaluate(time)`, this can ignore clone and save one field.